src/main/java/com/google/sps/servlets/TaskListServlet.java

```diff
@@ -0,0 +1,136 @@
+
+package com.google.sps.servlets;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.services.tasks.model.Task;
+import com.google.api.services.tasks.model.TaskList;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.sps.model.AuthenticatedHttpServlet;
+import com.google.sps.model.AuthenticationVerifier;
+import com.google.sps.model.TasksClient;
+import com.google.sps.model.TasksClientFactory;
+import com.google.sps.model.TasksClientImpl;
+import com.google.sps.utility.JsonUtility;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet to handle posting a new tasklist */
+@WebServlet("/tasklists")
+public class TaskListServlet extends AuthenticatedHttpServlet {
+  private final TasksClientFactory tasksClientFactory;
+
+  /** Create servlet with default TasksClientFactory and Authentication Verifier implementations */
+  public TaskListServlet() {
+    tasksClientFactory = new TasksClientImpl.Factory();
+  }
+
+  /**
+   * Create servlet with explicit implementations of TasksClientFactory and AuthenticationVerifier
+   *
+   * @param authenticationVerifier implementation of AuthenticationVerifier
+   * @param tasksClientFactory implementation of GmailClientFactory
+   */
+  public TaskListServlet(
+      AuthenticationVerifier authenticationVerifier, TasksClientFactory tasksClientFactory) {
+    super(authenticationVerifier);
+    this.tasksClientFactory = tasksClientFactory;
+  }
+
+  /**
+   * Request to get all taskLists with their associated tasks
+   *
+   * @param request HTTP request from client.
+   * @param response Http response to be sent to client. Contains a list of tasklists as well as a
+   *     list of tasks mapped to their parent tasklist id.
+   * @param googleCredential valid, verified google credential object
+   * @throws IOException if there's an issue with the tasks service
+   */
+  @Override
+  public void doGet(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    assert googleCredential != null
+        : "Null credentials (i.e. unauthenticated requests) should already be handled";
+
+    TasksClient tasksClient = tasksClientFactory.getTasksClient(googleCredential);
+
+    List<TaskList> taskLists = tasksClient.listTaskLists();
+    HashMap<String, List<Task>> taskListsWithTasks = mapTaskListsToTasks(taskLists, tasksClient);
+
+    Gson gson = new Gson();
+    JsonObject jsonObject = new JsonObject();
+    jsonObject.add("tasklists", gson.toJsonTree(taskLists));
+    jsonObject.add("tasks", gson.toJsonTree(taskListsWithTasks));
+    String json = gson.toJson(jsonObject);
+
+    response.setContentType("application/json");
+    response.getWriter().println(json);
```

Comment:
I think you can use JsonUtility here as well.

Reply:
Good idea - I got thrown off by using JsonElement. Thanks!

Reply:
I unfortunately cannot - gson needs to know that the object passed is of type `JsonObject` as it applies different serialization methods to that type. I logged an issue to extend the JsonUtility at a later date (#107)

Reply:
Alright, fair enough! Thanks!